src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -1973,71 +1974,115 @@ void Player_ZeroRootLimbYaw(Player* this) {
     this->skelAnime.jointTable[1].y = 0;
 }
 
-void func_80832DBC(Player* this) {
+/**
+ * Finishes "AnimMovement" by resetting various aspects of Player's SkelAnime structure.
+ *
+ * This function is called in Player_SetupAction so it will run on every action change, but
+ * it can also be called within action functions to change animations in the middle of an action.
+ */
+void Player_FinishAnimMovement(Player* this) {
     if (this->skelAnime.movementFlags != 0) {
-        func_808322FC(this);
+        Player_ApplyYawFromAnim(this);
+
         this->skelAnime.jointTable[0].x = this->skelAnime.baseTransl.x;
         this->skelAnime.jointTable[0].z = this->skelAnime.baseTransl.z;
+
         if (this->skelAnime.movementFlags & ANIM_FLAG_ENABLE_MOVEMENT) {
             if (this->skelAnime.movementFlags & ANIM_FLAG_UPDATE_Y) {
                 this->skelAnime.jointTable[0].y = this->skelAnime.prevTransl.y;
             }
         } else {
             this->skelAnime.jointTable[0].y = this->skelAnime.baseTransl.y;
         }
-        Player_SkelAnimeResetPrevTranslRot(this);
+
+        Player_ResetAnimMovement(this);
+
         this->skelAnime.movementFlags = 0;
     }
 }
 
-void func_80832E48(Player* this, s32 flags) {
-    Vec3f pos;
+/**
+ * This is a reimplementation of `AnimTask_ActorMovement`.
+ *
+ * This variation of AnimMovement will apply age-specific scaling to Player's world position
+ * in addition to his skeleton.
+ *
+ * When using the SkelAnime Anim Task implementation with `Player_StartAnimMovement` the
+ * age specific scaling will only be applied visually to Player's skeleton.
```

Comment:
what does it mean for it to be only applied "visually"?

Reply:
I suppose this sentence in general is pretty redundant with what is said above.
What if I change the comment as a whole to:
```
/**
 * This is a reimplementation of `AnimTask_ActorMovement`.
 *
 * This achieves the same goal as `AnimTask_ActorMovement` but it adds the ability to scale the resulting movement according to age.
 * When using the AnimTask variant, age specific scaling can only be applied visually to the root bone position and does not affect world position.
 ```

Reply:
Well I like the way this sounds more anyway so I pushed it. Not sure if it fixes the problem you had with it though.